hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSAdmin.java

```diff
@@ -2068,51 +2068,51 @@ int getReconfigurationStatus(final String nodeType, final String address, final
       errMsg = String.format("Node [%s] reloading configuration: %s.", address,
           e.toString());
     }
-
-    if (errMsg != null) {
-      err.println(errMsg);
-      return 1;
-    } else {
-      out.print(outMsg);
-    }
-
-    if (status != null) {
-      if (!status.hasTask()) {
-        out.println("no task was found.");
-        return 0;
-      }
-      out.print("started at " + new Date(status.getStartTime()));
-      if (!status.stopped()) {
-        out.println(" and is still running.");
-        return 0;
+    synchronized (this) {
```

Comment:
@slfan1989 This is good for concurrency control but we should avoid this for performance issues. Rather, we can update test.

Reply:
thanks for your suggestion, I will modify the junit test.